File: typescript/raindrops/raindrops.ts
export const convert = (x: number): string => {
  const drops: [number, string][] = [[3, "Pling"], [5, "Plang"], [7, "Plong"]]
  return drops
    .reduce((acc: string, [n, s]) => x % n === 0 ? acc + s : acc, "")
    || x.toString()
}

File: lua/raindrops/raindrops.lua
return function(n)
  local rain = ''

  if n % 3 == 0 then rain = rain..'Pling' end
  if n % 5 == 0 then rain = rain..'Plang' end
  if n % 7 == 0 then rain = rain..'Plong' end

  if rain == '' then return n..'' else return rain end
end
